src/app/thankYou/accountBenefits/accountBenefits.component.js

```diff
@@ -1,26 +1,51 @@
 import angular from 'angular';
-
 import sessionModalService from 'common/services/session/sessionModal.service';
-
+import sessionService from 'common/services/session/session.service';
 import template from './accountBenefits.tpl';
 
-let componentName = 'accountBenefits';
+import {Roles} from 'common/services/session/session.service';
 
-class AccountBenefitsController{
+let componentName = 'accountBenefits';
 
+class AccountBenefitsController {
   /* @ngInject */
-  constructor(sessionModalService){
+  constructor( sessionModalService, sessionService ) {
     this.sessionModalService = sessionModalService;
+    this.sessionService = sessionService;
+    this.hasUserMatch = false;
   }
 
+  $onChanges( changes ) {
+    // donorDetails is undefined initially
+    if ( changes.donorDetails && angular.isDefined( changes.donorDetails.currentValue ) ) {
+      // Show account benefits if registration state is NEW or MATCHED
+      this.hasUserMatch = this.donorDetails['registration-state'] !== 'COMPLETED';
```

Comment:
I think this variable name is confusing... this is just a flag to determine if user match stuff needs to be shown, not if it has a match or has been matched right? I would think if the account `hasUserMatch` or is `COMPLETED` this wouldn't need to be shown. The boolean value seems to be the opposite of the variable name.